### PR TITLE
Make remote sidecars non conflicting when several workspaces are running in the same k8s namespace

### PR DESCRIPTION
**Pull request by garagatyi**
### What does this PR do?
Make endpoint generation undeterminated process to avoid ports collision when several workspaces are running in the same k8s namespace.
Refactor and cleanup code.

### What issues does this PR fix or reference?
Needed for https://github.com/eclipse/che/issues/12238

### CODE REVIEW

File: brokers/theia/broker.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //
-// Copyright (c) 2012-2018 Red Hat, Inc.
+// Copyright (c) 2019 Red Hat, Inc.
```
Comment (sleshchenko) [inline brokers/theia/broker.go]: Should not license contain first contribution year and last modification year? Like `2018-2019`?

File: brokers/theia/broker.go [hunk 2]
```diff
@@ -224,13 +224,11 @@ func AddPortToTooling(toolingConf *model.ToolingConf, pj *PackageJSON) {
 	toolingConf.WorkspaceEnv = append(toolingConf.WorkspaceEnv, model.EnvVar{Name: theiaEnvVar1, Value: theiaEnvVarValue})
 }
 
-// GetRndNumberAsString returns stringified random port from range 4000-6000
-func GetRndNumberAsString() string {
-	port := GetRndPort()
+func (b *Broker) getRndNumberAsString() string {
+	port := b.getRndPort() // CHANge name to something meaningful and/OR random
```
Comment (sleshchenko) [inline brokers/theia/broker.go]: `CHANge` -> `Change`?
What is name in comment?
`meaningful and/OR random` <- looks like assigned value is just random port. 

So, please make sure comment is up to date.

File: brokers/vscode/broker.go [hunk 3]
```diff
@@ -213,3 +215,67 @@ func findAssetURL(response []byte, meta model.PluginMeta) (string, error) {
...
+		Name:       endpointName,
+		Public:     false,
+		TargetPort: port,
+	})
+	toolingConf.Containers[0].Env = append(toolingConf.Containers[0].Env, model.EnvVar{Name: "THEIA_PLUGIN_ENDPOINT_PORT", Value: sPort})
+	toolingConf.WorkspaceEnv = append(toolingConf.WorkspaceEnv, model.EnvVar{Name: theiaEnvVar1, Value: theiaEnvVarValue})
+}
+
+func (b *Broker) getRndNumberAsString() string {
+	port := b.getRndPort() // TODO CHANge name to something meaningful and/OR random
```
Comment (sleshchenko) [inline brokers/vscode/broker.go]: The same comment as in Theia plugin broker

File: brokers/vscode/broker.go [hunk 4]
```diff
@@ -213,3 +215,67 @@ func findAssetURL(response []byte, meta model.PluginMeta) (string, error) {
...
+// - Environment variable that start from THEIA_PLUGIN_REMOTE_ENDPOINT_ and ends with
+// plugin publisher and plugin name taken from packageJson and replacing all
+// chars matching [^a-z_0-9]+ with a dash character
+func (b *Broker) addPortToTooling(toolingConf *model.ToolingConf, pj *model.PackageJSON) {
+	port := b.getRndPort()
+	sPort := strconv.Itoa(port)
+	endpointName := "port" + sPort
+	var re = regexp.MustCompile(`[^a-zA-Z_0-9]+`)
+	prettyID := re.ReplaceAllString(pj.Publisher+"_"+pj.Name, `_`)
+	theiaEnvVar1 := "THEIA_PLUGIN_REMOTE_ENDPOINT_" + prettyID
```
Comment (sleshchenko) [inline brokers/vscode/broker.go]: `theiaEnvVar1` -> `theiaEnvVarKey`?

File: model/model.go [hunk 5]
```diff
@@ -179,3 +179,9 @@ type PluginBrokerLogEvent struct {
 
 // Type returns BrokerLogEventType.
 func (e *PluginBrokerLogEvent) Type() string { return BrokerLogEventType }
+
+// PackageJSON represents package.json file of JS based projects
+type PackageJSON struct {
+	Name      string  `json:"name" yaml:"name"`
+	Publisher string  `json:"publisher" yaml:"publisher"`
+}
```
Comment (sleshchenko) [inline model/model.go]: Please ass new line

File: brokers/theia/broker.go (hunk 1 above)
Comment (garagatyi) [reply brokers/theia/broker.go]: :+1:

File: brokers/theia/sidecar.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+// - Environment variable that start from THEIA_PLUGIN_REMOTE_ENDPOINT_ and ends with
+// plugin publisher and plugin name taken from packageJson and replacing all
+// chars matching [^a-z_0-9]+ with a dash character
+func addPortToTooling(toolingConf *model.ToolingConf, pj model.PackageJSON, rand common.Random) {
+	port := rand.IntFromRange(4000, 10000)
+	sPort := strconv.Itoa(port)
+	endpointName := rand.String(10)
+	var re = regexp.MustCompile(`[^a-zA-Z_0-9]+`)
+	prettyID := re.ReplaceAllString(pj.Publisher+"_"+pj.Name, `_`)
+	theiaEnvVar1 := "THEIA_PLUGIN_REMOTE_ENDPOINT_" + prettyID
```
[the remark this answers is not in the document]
Comment (garagatyi) [reply brokers/theia/sidecar.go]: :+1:

File: model/model.go (hunk 5 above)
Comment (garagatyi) [reply model/model.go]: :+1:

**Pull request merged**